byzcoin/contracts/coins.go

```diff
@@ -58,7 +58,16 @@ func (c *contractCoin) Spawn(rst byzcoin.ReadOnlyStateTrie, inst byzcoin.Instruc
 
 	// Spawn creates a new coin account as a separate instance.
 	ca := inst.DeriveID("")
-	log.Lvlf3("Spawning coin to %x", ca.Slice())
+	if pub := inst.Spawn.Args.Search("public"); pub != nil {
+		h := sha256.New()
+		h.Write([]byte(ContractCoinID))
+		h.Write(pub)
```

Comment:
Yes, you can coin squat here. But there are 2**256 coin addresses, so have fun. And if somebody did squat your coin, you take a next address. Anyway, this is why I use a `Spawner` contract where you have to pay some coins to create a coin address, to avoid somebody simply spawning coins over and over.

With regard to adding the length here, I don't see what attack this would prevent. The full input is chosable by the user, except for the domain-string `ContractCoinID`. I cannot come up with an attack in this case, but I'd be happy to be proven wrong.

Most attacks I heard of manage to create the same hash where it should not be possible. But due to the `ContractCoinID` this is not possible here.